WAZO-777 enable wazo-ui or create an index.html

fredericlepied opened the pull request.

fredericlepied committed:
add the possibility to enable wazo-ui

sduthil commented on roles/uc-ui/tasks/main.yml:
```diff
@@ -0,0 +1,62 @@
+- name: Configure ui
+  template:
+    src: templates/config.yml.j2
+    dest: /etc/wazo-ui/config.yml
```
`/etc/wazo-ui/config.yml` should be provided by the Debian package. Additional config should be added in `/etc/wazo-ui/conf.d/*.yml`.

fredericlepied replied:
done

sduthil commented on zuul.d/run.yaml:
```diff
@@ -22,3 +31,8 @@
+
+    - name: Check ui service
+      become: yes
+      shell: "systemctl status wazo-ui"
```
How about:

```
- name: Get service facts
  service_facts:
- name: Assert wazo-ui is running
  assert:
    that: "ansible_facts.services['zuul-gateway'].state == 'running'"
```

fredericlepied replied:
done

fredericlepied committed:
do not conflict with package config and make zuul validation more ansiblish

fredericlepied committed:
install an index.html when uc-ui is not enabled

fblackburn1 commented on roles/uc-ui/tasks/main.yml:
```diff
@@ -0,0 +1,62 @@
+    mode: 0640
+  when: engine_api_https_cert is defined
+  notify:
+    - restart wazo-ui
```
Why notify wazo-ui, when it is not installed?

fredericlepied replied:
when we execute uc-ui/tasks/main.yml, wazo-ui is always installed

fblackburn1 commented on roles/uc-ui/tasks/main.yml:
```diff
@@ -0,0 +1,62 @@
+    mode: 0640
+  when: engine_api_https_private_key is defined
+  notify:
+    - restart wazo-ui
```
same comment as above

fredericlepied replied:
sale as above

fblackburn1 commented on roles/uc-ui/templates/config.yml.j2:
```diff
@@ -0,0 +1,85 @@
+  host: {{ engine_api_host }}
+  port: {{ engine_api_port }}
+  prefix: /api/auth
+  verify_certificate: false
```
Do not ship default config with `verify_certificate `to False !

fredericlepied replied:
we need to discuss this one because it will not work if the ui is on a separate host

fblackburn1 commented on roles/uc-ui/templates/config.yml.j2:
```diff
@@ -0,0 +1,85 @@
+  prefix: /api/confd
+  verify_certificate: false
+
+policies:
```
remove unused section

fredericlepied replied:
ok

fblackburn1 commented on roles/uc-ui/templates/config.yml.j2:
```diff
@@ -0,0 +1,85 @@
+  customer:
+  location:
+
+enabled_plugins:
```
Since is already the default config, remove it

fredericlepied replied:
ok

fblackburn1 commented on roles/uc-ui/templates/config.yml.j2:
```diff
@@ -0,0 +1,85 @@
+  listen: {{ uc_ui_listen_address }}
+  port: {{ uc_ui_port}}
+  certificate: /usr/share/xivo-certs/server.crt
+  private_key: /usr/share/xivo-certs/server.key
```
Since it's the same value than the default, remove it

fredericlepied replied:
ok

fblackburn1 commented on roles/uc-ui/templates/config.yml.j2:
```diff
@@ -0,0 +1,85 @@
+
+https:
+  listen: {{ uc_ui_listen_address }}
+  port: {{ uc_ui_port}}
```
I don't know if we want to have custom listen address and port. I such a pain to migrate after ...

fredericlepied replied:
ok

fblackburn1 commented on zuul.d/run.yaml:
```diff
@@ -22,3 +31,11 @@
+
+    - name: Get service facts
+      become: yes
+      service_facts:
```
the empty key is needed?

fredericlepied replied:
yes that's the mandatory type of the task

fblackburn1 commented on zuul.d/run.yaml:
```diff
@@ -1,6 +1,15 @@
+      become: yes
+      blockinfile:
+        path: "{{ zuul.project.src_dir }}/inventories/uc-engine"
+        marker: "# Uncomment the 2 following lines if you want to deploy wazo-ui too:"
```
I don't know what is the good practice, but it seems a little bit weak to have a marker to a comment that doesn't deserve to be a marker

fredericlepied replied:
it was to edit like a user would do it...

fblackburn1 commented on roles/uc-ui/defaults/main.yml:
```diff
@@ -0,0 +1,2 @@
+uc_ui_listen_address: 0.0.0.0
+uc_ui_port: 9296
```
Do you have any use case for this?

fredericlepied replied:
No it was just mimicking the other services so removing it for now.

fblackburn1 reviewed (changes requested):
I'm not familiar with ansible, so sorry for some naive questions.
All default wazo-ui configuration should be removed and not duplicated. It is so horrible to migrate after having a drop-in config file.
Also, I'm not sure about the use case to allow custom values for listening address and port. If we do not have use case, I would prefer to avoid it and remove the custom config file completely.

fredericlepied committed:
clean config file and config vars according to the review

fblackburn1 committed:
remove wazo-ui config file

reason: for now, there are no real reason to make it configurable and
it only complicates future upgrades
